lib/cocoapods/installer/analyzer.rb

```diff
@@ -438,6 +438,18 @@ def generate_targets(resolver_specs_by_target, target_inspections)
           end
           target.search_paths_aggregate_targets.concat(search_paths_aggregate_targets).freeze
         end
+
+        aggregate_targets.each do |aggregate_target|
+          is_app_extension = !(aggregate_target.user_targets.map(&:symbol_type) &
+              [:app_extension, :watch_extension, :watch2_extension, :tv_extension, :messages_extension]).empty?
+          is_app_extension ||= aggregate_target.user_targets.any? { |ut| ut.common_resolved_build_setting('APPLICATION_EXTENSION_API_ONLY') == 'YES' }
+
+          next unless is_app_extension
+
+          aggregate_target.mark_application_extension_api_only
```

Comment:
Can the logic for determining `is_app_extension` live inside `AggregateTarget`. It looks like you're only using properties off `aggregate_target`. This allows you to remove `mark_application_extension_api_only` which is kinda ugly and get rid of the default `false` value.

Reply:
While the logic for `is_app_extension` can be put in AggregateTarget, `mark_application_extension_api_only` cannot be done away with - PodTarget also needs to be marked and it has no idea who its parent is.

Reply:
Ah bummer.

Reply:
I had a chat with @segiddins about it and it makes sense for the Analyzer to be the single one responsible for this weird flag.